### DISCUSSION TITLE
SIP Client to handle 302 to Invite

### DISCUSSION THREAD

**Discussion by michaeljdresser** [sipsorcery-org/sipsorcery#421] [General]
Hi, I'm trying to write a simple client using the SIPSorcery stack to initiate a SIP call, handle the 302 and send a new invite.

What would be the best way achieve this using SIPSorcery? Any help would be appreciated.

**Client (SIP Sorcery)**                 **SIP Server(s)**
Invite ------------------------>
          <-----------------------  302
ACK   ------------------------>
Invite ------------------------>
          <-----------------------  100 Trying
          <-----------------------  180 Ringing
          <-----------------------  200 OK
BYE     ------------------------>
          <-----------------------  200 OK

**Comment by sipsorcery** [comment]
In theory processing the redirect is as simple as completing the current call and starting a new one with the new destination provided from the Contact header in the redirect response.

In practice redirects, like transfers, can be a can of worms. In the older version of the code in this library there was a bit of a ceremony for [redirect processing](https://github.com/sipsorcery-org/sipsorcery/blob/9ecaafb996a9d5ab4deccb22fa241c8acf5f0ab5/sipsorcery-core/SIPSorcery.AppServer.DialPlan/DialPlanApps/ForkCall.cs#L502). The main concern is that a malicious party could redirect a caller to a premium rate number or some other dodgy destination.

If you trust the called party maybe you could assume the redirect will never be malicious and automatically process the response. Otherwise at the very least it may be better to display some kind of confirmation prompt to the user.
